conans/test/functional/generators/cmake_find_package_test.py

```diff
@@ -430,7 +433,9 @@ def build(self):
         client.run("build .")
         self.assertIn('Found MYHELLO2: 1.0 (found version "1.0")', client.out)
         self.assertIn('Found MYHELLO: 1.0 (found version "1.0")', client.out)
-        self.assertIn("Target libs: CONAN_LIB::MYHELLO2_hello;;;CONAN_LIB::MYHELLO_hello",
+        self.assertIn("Target libs (hello2): CONAN_LIB::MYHELLO2_hello;MYHELLO::MYHELLO;MYHELLO::MYHELLO;;",
```

Comment:
Repeated MYHELLO::MYHELLO, is this OK?

Reply:
This is probably something we can (and should) improve, but I think there is no risk to add it twice: duplicated header folders, libraries, but it is harmless. Removing it will need some _wonderful_ CMake code.

---
As a side note, I think we should do more work on the python side with the `cpp_info` and `deps_cpp_info` before rendering the generators, that way we will do less work on the CMake side which is harder to read and harder to maintain. That perspective will probably be helpful for the components feature.

Reply:
Agree with both things. More logic should be done in the python side. Also, the above duplication, I am  not concerned about the duplication itself in the output, but understanding where it comes from and that it is not an underlying bug